ansible/roles_ocp_workloads/ocp4_workload_quarkus_super_heroes_demo/tasks/deploy_cicd.yaml

```diff
@@ -0,0 +1,270 @@
+  include_tasks: install_prometheus.yml
+  when: ocp4_workload_quarkus_super_heroes_demo_install_prometheus | bool
+
+- name: "[{{ t_project_name_test }}] - install Grafana"
+  vars:
+    t_project_name: "{{ t_project_name_test }}"
+  include_tasks: install_grafana.yml
+  when: ocp4_workload_quarkus_super_heroes_demo_install_grafana | bool
+
+- name: "[{{ t_project_name_test }}] - create imagestreams"
+  kubernetes.core.k8s:
+    state: present
+    api_version: image.openshift.io/v1
+    kind: ImageStream
+    definition:
+      metadata:
+        name: "{{ item }}"
+        namespace: "{{ t_project_name_test }}"
+  loop:
+  - rest-fights
+
+- name: "[{{ t_project_name_test }}] - create buildconfigs"
+  kubernetes.core.k8s:
+    state: present
+    definition: "{{ lookup('template', 'tekton/buildconfig.yaml.j2') }}"
+  loop:
+  - rest-fights
+
+- name: "[{{ t_project_name_pipelines }}] - create project"
+  kubernetes.core.k8s:
+    state: present
+    api_version: project.openshift.io/v1
+    kind: Project
+    name: "{{ t_project_name_pipelines }}"
+    definition:
+      metadata:
+        annotations:
+          openshift.io/description: "Quarkus Superheroes Pipelines"
+          openshift.io/display-name: "Quarkus Superheroes Pipelines"
+        name: "{{ t_project_name_pipelines }}"
+
+- name: create gitea token
+  ansible.builtin.uri:
+    # yamllint disable-line rule:line-length
+    url: "{{ r_gitea.resources[0].status.giteaRoute }}/api/v1/users/{{ ocp4_workload_quarkus_super_heroes_demo_clone_repo_gitea_user }}/tokens"
+    method: POST
+    force_basic_auth: true
+    user: "{{ ocp4_workload_quarkus_super_heroes_demo_clone_repo_gitea_user }}"
+    password: "{{ ocp4_workload_quarkus_super_heroes_demo_clone_repo_gitea_password }}"
+    return_content: true
+    headers:
+      accept: application/json
+    body_format: json
+    body: |
+      {
+        "name": "pipeline_tkn_{{ 1000 | random }}"
+      }
+    status_code: [201]
+  register: r_gitea_token
+
+- name: "[{{ t_project_name_pipelines }}] - create secret gitea-token"
+  vars:
+    _gitea_user_url: "{{ r_gitea.resources[0].status.giteaRoute }}/{{ ocp4_workload_quarkus_super_heroes_demo_clone_repo_gitea_user }}"
+    _gitea_token_username: "{{ r_gitea_token.json.name }}"
+    _gitea_token_password: "{{ r_gitea_token.json.sha1 }}"
+  kubernetes.core.k8s:
+    state: present
+    definition: "{{ lookup('template', 'tekton/secret-gitea-token.yaml.j2') }}"
+
+- name: "[{{ t_project_name_pipelines }}] - link gitea-token service account pipeline"
+  kubernetes.core.k8s:
+    state: patched
+    kind: ServiceAccount
+    name: pipeline
+    namespace: "{{ t_project_name_pipelines }}"
+    definition:
+      secrets:
+      - name: gitea-token
+
+- name: "[{{ t_project_name_pipelines }}] - create tekton tasks"
+  kubernetes.core.k8s:
+    state: present
+    definition: "{{ lookup('template', 'tekton/{{ item }}.yaml.j2') }}"
+  loop:
+  - pvc
```

Comment:
Interesting approach. I usually write the whole file name in the loop.

Reply:
Will do.